libevm/VMCalls.cpp

```diff
@@ -248,27 +250,26 @@ bool VM::caseCallSetup(CallParameters *callParams, bytesRef& o_output)
 	ON_OP();
 	updateIOGas();
 
-	if (m_OP != Instruction::DELEGATECALL && m_SP[2] > 0)
+	if (haveValueArg && m_SP[2] > 0)
 		callParams->gas += m_schedule->callStipend;
 
 	callParams->codeAddress = destinationAddr;
 
-	unsigned inOutOffset = 0;
-	if (m_OP == Instruction::DELEGATECALL || m_OP == Instruction::STATICCALL)
+	if (haveValueArg)
 	{
-		callParams->apparentValue = m_ext->value;
-		callParams->valueTransfer = 0;
+		callParams->valueTransfer = m_SP[2];
+		callParams->apparentValue = m_SP[2];
 	}
 	else
 	{
-		callParams->apparentValue = callParams->valueTransfer = m_SP[2];
-		inOutOffset = 1;
+		callParams->apparentValue = m_ext->value;
```

Comment:
All of them are quite complex. We need a simple test with a transaction having no-zero value, then staticcall and check of the callvalue inside this staticcall.

Similar one with a CALL instead of transaction.